packages/flutter/lib/src/cupertino/action_sheet.dart

```diff
@@ -0,0 +1,1264 @@
+
+  /// The callback that is called when the button is tapped.
+  ///
+  /// This attribute must not be null.
+  final VoidCallback onPressed;
+
+  /// Whether this action is the default choice in the action sheet.
+  ///
+  /// Default buttons have bold text.
+  final bool isDefaultAction;
+
+  /// Whether this action might change or delete data.
+  ///
+  /// Destructive buttons have red text.
+  final bool isDestructiveAction;
+
+  /// The widget below this widget in the tree.
+  ///
+  /// Typically a [Text] widget.
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    TextStyle style = _kActionSheetActionStyle;
+
+    if (isDefaultAction) {
+      style = style.copyWith(fontWeight: FontWeight.w600);
+    }
+
+    if (isDestructiveAction) {
+      style = style.copyWith(color: CupertinoColors.destructiveRed);
+    }
+
+    return new GestureDetector(
+      onTap: onPressed,
+      behavior: HitTestBehavior.opaque,
+      child: new ConstrainedBox(
+        constraints: const BoxConstraints(
+          minHeight: _kButtonHeight,
+        ),
+        child: new Semantics(
+          button: true,
+          child: new Container(
+            alignment: Alignment.center,
+            padding: const EdgeInsets.symmetric(
+              vertical: 16.0,
+              horizontal: 10.0,
+            ),
+            child: new DefaultTextStyle(
+              style: style,
+              child: child,
+              textAlign: TextAlign.center,
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class _CupertinoActionSheetCancelButton extends StatefulWidget {
+  const _CupertinoActionSheetCancelButton({
+    Key key,
+    this.child,
+  }) : super(key: key);
+
+  final Widget child;
+
+  @override
+  _CupertinoActionSheetCancelButtonState createState() => _CupertinoActionSheetCancelButtonState();
+}
+
+class _CupertinoActionSheetCancelButtonState extends State<_CupertinoActionSheetCancelButton> {
+  Color _backgroundColor;
+
+  @override
+  void initState() {
+    _backgroundColor = CupertinoColors.white;
+    super.initState();
+  }
+
+  void _onTapDown(TapDownDetails event) {
+    setState(() {
+      _backgroundColor = _kCancelButtonPressedColor;
+    });
+  }
+
+  void _onTapUp(TapUpDetails event) {
+    setState(() {
+      _backgroundColor = CupertinoColors.white;
+    });
+  }
+
+  void _onTapCancel() {
+    setState(() {
+      _backgroundColor = CupertinoColors.white;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new GestureDetector(
+      excludeFromSemantics: true,
+      onTapDown: _onTapDown,
+      onTapUp: _onTapUp,
+      onTapCancel: _onTapCancel,
+      child: new Container(
+        decoration: new BoxDecoration(
+          color: _backgroundColor,
+          borderRadius: BorderRadius.circular(_kCornerRadius),
+        ),
+        child: widget.child,
+      ),
+    );
+  }
+}
+
+class _CupertinoAlertRenderWidget extends RenderObjectWidget {
```

Comment:
I'm assuming everything here below are things copied from @matthew-carroll's PR?

Reply:
Yes everything below is copied, with some modifications to make things work for action sheets specifically